test/unit/scripts/upgrade/TimeConstraints.t.sol

```diff
@@ -34,97 +34,182 @@ contract TimeConstraintsTest is Test {
-            )
-        );
+    // checkTimeWithinDayTimeAndEmit
+
+    function testFuzz_checkTimeWithinDayTimeAndEmit_HappyPath_Regular(
+        Duration from,
+        Duration to,
+        Duration timeToWarp
+    ) external {
+        vm.assume(to < timeConstraints.DAY_DURATION());
+        vm.assume(from <= timeToWarp && timeToWarp <= to);
+        vm.warp(timeToWarp.toSeconds());
+        vm.expectEmit();
+        emit TimeConstraints.TimeWithinDayTimeChecked(from, to);
+        timeConstraints.checkTimeWithinDayTimeAndEmit(from, to);
+    }
 
-        timeConstraints.checkExecuteWithinDayTime(Durations.from(18 hours), Durations.from(12 hours));
+    function testFuzz_checkTimeWithinDayTimeAndEmit_HappyPath_Overnight(
+        Duration from,
+        Duration to,
+        Duration timeToWarp
+    ) external {
+        vm.assume(from < timeConstraints.DAY_DURATION() && to < from);
+        vm.assume(timeToWarp <= to || timeToWarp >= from);
+        vm.assume(timeToWarp < timeConstraints.DAY_DURATION());
+        vm.warp(timeToWarp.toSeconds());
+        vm.expectEmit();
+        emit TimeConstraints.TimeWithinDayTimeChecked(from, to);
+        timeConstraints.checkTimeWithinDayTimeAndEmit(from, to);
     }
 
-    function testFuzz_checkExecuteAfterTimestamp_HappyPath(uint32 secondsAfter) public {
-        vm.warp(BASE_TIMESTAMP + secondsAfter);
-        timeConstraints.checkExecuteAfterTimestamp(Timestamps.from(BASE_TIMESTAMP));
+    // checkTimeAfterTimestamp
+
+    function testFuzz_checkTimeAfterTimestamp_HappyPath(Timestamp timestampToWarp, Timestamp timestampToCheck) public {
+        vm.assume(timestampToWarp > timestampToCheck);
+        vm.warp(timestampToWarp.toSeconds());
+        timeConstraints.checkTimeAfterTimestamp(timestampToCheck);
     }
 
-    function testFuzz_checkExecuteAfterTimestamp_NotReached(uint32 secondsBefore) public {
-        vm.assume(secondsBefore < BASE_TIMESTAMP && secondsBefore > 0);
+    function testFuzz_checkTimeAfterTimestamp_NotReached(
+        Timestamp timestampToWarp,
+        Timestamp timestampToCheck
+    ) public {
+        vm.assume(timestampToWarp <= timestampToCheck);
 
-        vm.warp(BASE_TIMESTAMP - secondsBefore);
+        vm.warp(timestampToWarp.toSeconds());
         vm.expectRevert(
-            abi.encodeWithSelector(TimeConstraints.TimestampNotReached.selector, Timestamps.from(BASE_TIMESTAMP))
+            abi.encodeWithSelector(TimeConstraints.TimestampNotReached.selector, timestampToCheck.toSeconds())
         );
-        timeConstraints.checkExecuteAfterTimestamp(Timestamps.from(BASE_TIMESTAMP));
+        this.external__checkTimeAfterTimestamp(timestampToCheck);
     }
 
-    function test_checkExecuteBeforeTimestamp_HappyPath() public {
-        vm.warp(BASE_TIMESTAMP);
-        timeConstraints.checkExecuteBeforeTimestamp(Timestamps.from(BASE_TIMESTAMP));
+    // checkTimeAfterTimestampAndEmit
+
+    function testFuzz_checkTimeAfterTimestampAndEmit_HappyPath(
```

Comment:
Add the same set of negative tests for the "Emit" methods as for the "regular" check methods.